pallets/merkle/src/merkle/helper.rs

```diff
@@ -15,16 +15,23 @@ pub struct ZkProof {
 	pub leaf_com: Commitment,
 	pub path: Vec<(Commitment, Commitment)>,
 	pub r_com: Commitment,
-	pub nullifier: Data,
+	pub nullifier_com: Commitment,
+	pub nullifier_hash: Data,
 	pub bytes: Vec<u8>,
 }
 
+/// Constrain a linear combination to be equal to a scalar
+pub fn constrain_lc_with_scalar<CS: ConstraintSystem>(cs: &mut CS, lc: LinearCombination, scalar: &Scalar) {
```

Comment:
If we opt to use this function, then we should use it everywhere, but it doesn't make our life terribly easier

Reply:
Agreed, I may end up swapping our gadgets this week so we do use this in gadgets lib.